### ISSUE TITLE
Gracefully parse `abstract class` in expression position

### ISSUE THREAD

**Issue by mattpap** [microsoft/TypeScript#17416]
**TypeScript Version:**  2.4.2 / nightly (2.5.0-dev.20170725)

**Code**

```ts
function f1() {
  return abstract class {}
}

function f2() {
  const x = abstract class {}
  return x
}

function f3() {
  abstract class x {}
  return x
}
```

**Expected behavior:**

No errors reported by the compiler.

**Actual behavior:**

```
some.ts(2,19): error TS1005: ';' expected.
some.ts(6,22): error TS1005: ',' expected.
```

**Comment by DanielRosenwasser** [comment]
We aren't looking to support them (see #4578), but it wouldn't hurt to parse these and provide a useful error.

**Comment by SamuraiJack** [comment]
Please implement this, simplifies notation, when using mixins.
